model/authentication.go

```diff
@@ -83,3 +87,16 @@ func (auth *Authentication) ToVaultMap() (map[string]interface{}, error) {
 	// Vault requires the hash to be wrapped in a "data" object in order to be accepted.
 	return map[string]interface{}{"data": data}, nil
 }
+func (auth *Authentication) UpdateBy(attributes map[string]interface{}) error {
+	data, err := json.Marshal(attributes)
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(data, &auth)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
I know marhshaling + unmarshaling might be the nicest way to do this here - but I really think we should just do a loop over the map and update the fields. Much like we do for [the `UpdateFromRequest` methods on various models](https://github.com/RedHatInsights/sources-api-go/blob/7790b0e4d534d2ed510c7cd4470f77b4022b6704/model/authentication_http.go#L47-L69). It looks like something similar was done in the `authFromVault()` method too at the top of this PR.

This way it is explicit what we are doing and we don't have to worry about something going wrong with either of the json marshals.